base/v0_3_exp/translate.go

```diff
@@ -387,7 +423,21 @@ func (c Config) addMountUnits(ret *types.Config) translate.TranslationSet {
 			continue
 		}
 		fromPath := path.New("yaml", "storage", "filesystems", i, "with_mount_unit")
-		newUnit := mountUnitFromFS(fs)
+		remote := false
+		// check filesystems targeting /dev/mapper devices against LUKS to determine if a
+		// remote mount is needed
+		if strings.HasPrefix(fs.Device, "/dev/mapper/") || strings.HasPrefix(fs.Device, "/dev/disk/by-id/dm-name-") {
+			for _, luks := range c.Storage.Luks {
+				// LUKS devices are opened with their name specified
+				deviceName := filepath.Base(fs.Device)
```

Comment:
Can be hoisted outside the loop.

As @LorbusChris noticed, we should be using `path` instead of `filepath` when we're referring to specifically Unix paths.  Otherwise we're expecting `C:\dev\mapper\my-luks-device` on Windows.

Reply:
Ended up dropping as a result of going with the other suggestion for checking the entire device path rather than just the device name.